functions/Get-DbaHelpIndex.ps1

```diff
@@ -0,0 +1,596 @@
+	If not passed it will return data on all indexes in a given database.
+	
+.PARAMETER IncludeStats
+This includes statistics as well as indexes in the output (statistics information such as the StatsRowMods will always be returned for indexes).
+	
+.PARAMETER IncludeDataTypes
+This will include the data type of each column that makes up a part of the index definition (key and include columns)
+	
+.PARAMETER FormatResults
+Returns the numerical data in a more user readable format (numerical separator will depend on localization settings).
+	
+
+.NOTES 
+Original Author: Nic Cain, https://sirsql.net/
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.	
+
+.LINK
+https://dbatools.io/Get-DbaHelpIndex
+
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Databases MyDB
+	Returns information on all indexes on the MyDB database on the localhost.
+
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Databases MyDB,MyDB2
+	Returns information on all indexes on the MyDB, MyDB2 databases
+	
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Databases MyDB -ObjectName dbo.Table1
+	Returns index information on the object dbo.Table1 in the database MyDB
+	
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Databases MyDB -ObjectName dbo.Table1 -IncludeStats
+	Returns information on the indexes and statistics for the table dbo.Table1 in the MyDB database
+	
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Database MyDB -ObjectName dbo.Table1 -IncludeDataTypes
+	Returns the index information for the table dbo.Table1 in the MyDB database, and includes the data types for the key and include columns
+	
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Database MyDB -ObjectName dbo.Table1 -FormatResults
+	Returns the index information for the table dbo.Table1 in the MyDB database, and returns the numerical data with separators to make it more readable (ie 1234 becomes 1,234)
+	
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Database MyDB -IncludeStats -FormatResults
+	Returns the index information for all index in the MyDB database, as well as statistics, and formats the numerical data to be more redable	
+	
+#>
+	
+	[CmdletBinding(SupportsShouldProcess = $true)]
```

Comment:
I don't think it does support shouldprocess.
-Whatif or -Confirm have no effect, I just get output straight away.